homeassistant/components/weather/__init__.py

```diff
@@ -221,6 +222,38 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         ],
         supports_response=SupportsResponse.ONLY,
     )
+    component.async_register_entity_service(
+        SERVICE_GET_FORECAST_ATTR,
+        {
+            vol.Required("attribute"): vol.In(
+                (
+                    ATTR_FORECAST_CONDITION,
+                    ATTR_FORECAST_HUMIDITY,
+                    ATTR_FORECAST_PRECIPITATION,
+                    ATTR_FORECAST_PRECIPITATION_PROBABILITY,
+                    ATTR_FORECAST_PRESSURE,
+                    ATTR_FORECAST_APPARENT_TEMP,
+                    ATTR_FORECAST_TEMP,
+                    ATTR_FORECAST_TEMP_LOW,
+                    ATTR_FORECAST_WIND_BEARING,
+                    ATTR_FORECAST_WIND_GUST_SPEED,
+                    ATTR_FORECAST_WIND_SPEED,
+                    ATTR_FORECAST_DEW_POINT,
+                    ATTR_FORECAST_CLOUD_COVERAGE,
+                    ATTR_FORECAST_UV_INDEX,
+                )
+            ),
+            vol.Optional("limit_hours"): vol.Range(1, 24 * 31),
+            vol.Optional("min_max", default="max"): vol.In(("min", "max")),
```

Comment:
Min max isn't used.

Reply:
I also still think the user should be able to select which forecast to pick from + limit_hours should be renamed to be a fit for all three types.

Reply:
@gjohansson-ST Right, I think time_limit is a better name. 

I don't see any use case to let the user pick a forecast type. Furthermore, this would blow up this PR completely.
The result would be the same between
```
forecast_type: hourly
time_limit: 24h
```
and
```
forecast_type: daily
time_limit: 1d
```

Reply:
No? A 24 hour window for hourly forecast would return a list of 24 items while for daily it would return 1.

Reply:
I see. Basically your idea is to extend the service `weather.get_forecast` with a time limit and attribute filter, right?

Reply:
Right, should return a list `[11, 12.1, 11.3, ...]` as example for the attribute selected and the values within the timeframe.
That would then be very simple for a user to use in a template to calculate max, min or whatever that is wished for.

Reply:
I am ok with that. My original purpose is superseded because we do not perform calculations in services.
Would it be a better way to extend the existing service with optional parameters?

Reply:
No, should be a separate service